packages/ra-core/src/reducer/admin/resource/list/total.js

```diff
@@ -8,6 +9,9 @@ export default resource => (previousState = 0, { type, payload, meta }) => {
     if (!meta || meta.resource !== resource) {
         return previousState;
     }
+    if (type === CRUD_GET_ONE_SUCCESS) {
+        return previousState == 0 ? 1 : previousState;
```

Comment:
This is to avoid the "no results found" after a redirection from the Edit view if the user didn't display the list before. The bug was an artefact of optimistic rendering.